### PR TITLE
Unwrap InvocationTargetException thrown by QuarkusTest*Callback

### PR DESCRIPTION
**Pull request by famod**
Improves readability when you have e.g. a custom callback that validates something.

Some methods are only allowed to throw `Exeption`, not `Throwable`, hence the cause type check.

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java [hunk 1]
```diff
@@ -504,16 +506,18 @@ public void afterEach(ExtensionContext context) throws Exception {
...
-                Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
-                afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
-                        .invoke(afterEachCallback, tuple.getValue());
-            }
             try {
+                for (Object afterEachCallback : afterEachCallbacks) {
+                    Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
+                    afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
+                            .invoke(afterEachCallback, tuple.getValue());
+                }
```
Comment (famod) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:514]: Running this outside of the TCCL restoration block looked wrong, so I moved it inside.

**Comment by famod** [comment]
I'll add the backport label because I think that TCCL thing is a bug (rare though).

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java (hunk 1 above)
Comment (famod) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:514]: Not an actual one, but compare it with this:
https://github.com/quarkusio/quarkus/blob/2.7.2.Final/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L407

Why should the TCCL only be restored if there is an exception in a beforeEachCallback, but _not_ if there is an exception in an afterEachCallback? Doesn't make sense to me.

Comment (famod) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:514]: PS: afterAll also restores the TCCL just like beforeEach. Same happens for beforeClassCallback. 🤷‍♂️

**Review by geoand — APPROVED** [review]

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java (hunk 1 above)
Comment (stuartwdouglas) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:514]: This change looks right to me

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**